lib/Sema/CSApply.cpp

```diff
@@ -7593,6 +7593,12 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
       }
     }
 
+    if (fromEI.getLifetimeDependenceInfo() !=
```

Comment:
This problem should be detected during solving and diagnosed via a `ConstraintFix`

Reply:
In general we are trying to avoid diagnosing anything during CSApply because it means that the solution that solver produces is invalid…

Reply:
I moved it to `ConstraintSystem::matchFunctionTypes` which is handling similar errors